ios/RNSScreen.mm

```diff
@@ -964,6 +989,73 @@ - (void)viewDidLayoutSubviews
   }
 }
 
+- (void)dismissViewControllerAnimated:(BOOL)flag completion:(void (^)())completion
+{
+  [super dismissViewControllerAnimated:flag completion:completion];
+  BOOL isPresentedAsNativeModal = self.screenView.isPresentedAsNativeModal;
+
+  [self recalculateHeaderHeightIsModal:isPresentedAsNativeModal];
+}
+
+- (CGFloat)getCalculatedHeaderHeight:(BOOL)isModal
+{
+  CGFloat navbarHeight = self.navigationController.navigationBar.frame.size.height;
+
+  // In case where screen is a modal, we want to calculate it's childViewController's height
+  if (isModal && self.childViewControllers.count > 0 && self.childViewControllers[0] != nil) {
```

Comment:
I think something like this would be better since you cast it later.

```suggestion
  if (isModal && self.childViewControllers.count > 0 && [self.childViewControllers[0] isKindOfClass:UINavigationController.class]) {
```

Reply:
Also, what if we had double nested stack? Will it work correctly in such case?

Reply:
@WoLewicki I checked that and it seems to work already!